### PR TITLE
Nest component & key hotfix

### PR DESCRIPTION
**Pull request by Z33**
- Added Nest component
- Added Framer Motion
- Updated Platen.io docs
- Fixed multiple item key's

### CODE REVIEW

**Comment by michaeltlombardi** [comment]
Questions for you/ @Dastrdlydave:

1. For nested, is it possible to opt between having the child items be in a scrollable frame vs just unhidden?
2. Should it be possible to set a custom leading icon for the nest title? ie start with a dice or an exclamation or whatever?

Otherwise this looks awesome!

File: content/books/platen/chapters/3/index.mdx [hunk 1]
```diff
@@ -6,82 +6,142 @@ last-updated: 2020-06-09
...
 
+### Preview
+
 <SecLibrary />
 
-<Clip id="rolltable">
+</Nest>
 
-## Widget, Rolltable
+### Widgets
```
Comment (michaeltlombardi) [inline content/books/platen/chapters/3/index.mdx]: Should this be an H2 now as the nested bits are all larger/seem more important than this header which sits over them?

Or maybe the nested bits should be scaled down a little to be H3 or H4 sized? I don't think we can dynamically assume their size based on the last heading level in the doc but that would be the sweetest.

File: content/books/platen/chapters/3/index.mdx [hunk 2]
```diff
@@ -6,82 +6,142 @@ last-updated: 2020-06-09
 
 # Custom Components
 
-<Clip id="part-1">
+## Sections
 
-## Section, Library
+<Nest title="<Library />">
```
Comment (michaeltlombardi) [inline content/books/platen/chapters/3/index.mdx]: Here we imply we're renaming `SecLibrary` to `Library` but it still seems to be `SecLibrary` in the page?

**Comment by Z33** [comment]
> Questions for you/ @Dastrdlydave:
> 
> 1. For nested, is it possible to opt between having the child items be in a scrollable frame vs just unhidden?
> 2. Should it be possible to set a custom leading icon for the nest title? ie start with a dice or an exclamation or whatever?
> 
> Otherwise this looks awesome!

1. Yeah, I think having a max-height attribute would handle this. Two ways to handle this: Default set at 600px with "none" option OR Default is set to "none".
2. I'm looking into this for all widgets. This will be part of the markdown icon imports.

File: content/books/platen/chapters/3/index.mdx (hunk 1 above)
Comment (Z33) [reply content/books/platen/chapters/3/index.mdx]: I can scale down the typography inside the Nest's with the Tailwindcss prose-sm class if that's what you're asking.

File: content/books/platen/chapters/3/index.mdx (hunk 2 above)
Comment (Z33) [reply content/books/platen/chapters/3/index.mdx]: Correct, I'll resolve

**Comment by michaeltlombardi** [comment]
I'd be happy to merge then with a max-height attribute implemented with either default!

File: content/books/platen/chapters/3/index.mdx (hunk 1 above)
Comment (michaeltlombardi) [reply content/books/platen/chapters/3/index.mdx]: I think just the nest header/lead, maybe?

**Review by michaeltlombardi — APPROVED** [review]
Looks awesome!

**Pull request merged**